user-ops-indexer/user-ops-indexer-logic/src/indexer/rpc_utils.rs

```diff
@@ -1,9 +1,42 @@
-use async_trait::async_trait;
-use ethers::prelude::{
-    Action, Address, Bytes, CallFrame, CallType, GethDebugBuiltInTracerType, GethDebugTracerType,
-    GethDebugTracingOptions, GethTrace, GethTraceFrame, JsonRpcClient, Middleware, NodeClient,
-    Provider, ProviderError, TxHash,
+use alloy::{
+    primitives::{Address, Bytes, TxHash},
+    providers::{
+        ext::{DebugApi, TraceApi},
+        Provider,
+    },
+    rpc::types::trace::{
+        geth::{
+            CallFrame, GethDebugBuiltInTracerType, GethDebugTracerConfig, GethDebugTracerType,
+            GethDebugTracingOptions, GethDefaultTracingOptions, GethTrace,
+        },
+        parity::{Action, CallType},
+    },
+    transports::{TransportErrorKind, TransportResult},
 };
+use async_trait::async_trait;
+use serde::Deserialize;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Deserialize)]
+pub enum TraceClient {
+    Geth,
+    Parity,
+}
+
+impl From<String> for TraceClient {
+    fn from(value: String) -> Self {
+        if value.contains("Geth/") {
```

Comment:
didnt get it, why contains `Geth/` and not simple [derive_fromstr_from_deserialize](https://docs.rs/serde_plain/latest/serde_plain/macro.derive_fromstr_from_deserialize.html)

Reply:
Introduced a second enum with logic similar to the one from `ethers-rs` https://github.com/gakonst/ethers-rs/blob/6e2ff0ef8af8c0ee3c21b7e1960f8c025bcd5588/ethers-providers/src/rpc/provider.rs#L43-L69 . Seems like they removed it from alloy, unfortunately.